diff --git a/include/verilated.mk.in b/include/verilated.mk.in
index a909077..28716ed 100644
--- a/include/verilated.mk.in
+++ b/include/verilated.mk.in
@@ -62,3 +62,3 @@ CFG_LDLIBS_DYNAMIC = @CFG_LDLIBS_DYNAMIC@
-VERILATOR_COVERAGE = $(PERL) $(VERILATOR_ROOT)/bin/verilator_coverage
-VERILATOR_INCLUDER = $(PYTHON3) $(VERILATOR_ROOT)/bin/verilator_includer
-VERILATOR_CCACHE_REPORT = $(PYTHON3) $(VERILATOR_ROOT)/bin/verilator_ccache_report
+VERILATOR_COVERAGE = $(PERL) "$(VERILATOR_ROOT)/bin/verilator_coverage"
+VERILATOR_INCLUDER = $(PYTHON3) "$(VERILATOR_ROOT)/bin/verilator_includer"
+VERILATOR_CCACHE_REPORT = $(PYTHON3) "$(VERILATOR_ROOT)/bin/verilator_ccache_report"
@@ -72,7 +71,0 @@ export CCACHE_SLOPPINESS
-######################################################################
-# Make checks
-
-ifneq ($(words $(CURDIR)),1)
-  $(error Unsupported: GNU Make cannot build in directories containing spaces, build elsewhere: '$(CURDIR)')
-endif
-
@@ -81 +74,5 @@ endif
-UNAME_S := $(shell uname -s)
+ifeq ($(OS),Windows_NT)
+  UNAME_S := Windows_NT
+else
+  UNAME_S := $(shell uname -s)
+endif
@@ -89,2 +86,2 @@ VK_CPPFLAGS_ALWAYS += \
-  -I$(VERILATOR_ROOT)/include \
-  -I$(VERILATOR_ROOT)/include/vltstd \
+  -I"$(VERILATOR_ROOT)/include" \
+  -I"$(VERILATOR_ROOT)/include/vltstd" \
@@ -109,2 +106,2 @@ VPATH += ..
-VPATH += $(VERILATOR_ROOT)/include
-VPATH += $(VERILATOR_ROOT)/include/vltstd
+VPATH += $(VERILATOR_ROOT_MAKE)/include
+VPATH += $(VERILATOR_ROOT_MAKE)/include/vltstd
@@ -252,0 +250,5 @@ endif
+ifeq ($(OS),Windows_NT)
+%.a:
+	@if exist "$@" del /Q "$@"
+	$(AR) -rcs $@ $(filter %.o,$^)
+else
@@ -271,0 +274 @@ endif
+endif
@@ -298,2 +301,2 @@ ifneq ($(VM_DEFAULT_RULES),0)
-  $(VK_GLOBAL_OBJS): %.o: %.cpp
-	$(OBJCACHE) $(CXX) $(OPT_GLOBAL) $(CXXFLAGS) $(CPPFLAGS) -c -o $@ $<
+  $(VK_GLOBAL_OBJS): %.o: $(VERILATOR_ROOT_MAKE)/include/%.cpp
+	$(OBJCACHE) $(CXX) $(OPT_GLOBAL) $(CXXFLAGS) $(CPPFLAGS) -c -o $@ "$<"
diff --git a/src/V3EmitMk.cpp b/src/V3EmitMk.cpp
index a63d328..22630e1 100644
--- a/src/V3EmitMk.cpp
+++ b/src/V3EmitMk.cpp
@@ -656,0 +657,4 @@ public:
+        of.puts("empty :=\n");
+        of.puts("space := $(empty) $(empty)\n");
+        of.puts(
+            "VERILATOR_ROOT_MAKE := $(subst $(space),\\$(space),$(VERILATOR_ROOT))\n");
@@ -732 +736 @@ public:
-        of.puts("include $(VERILATOR_ROOT)/include/verilated.mk\n");
+        of.puts("include $(VERILATOR_ROOT_MAKE)/include/verilated.mk\n");
diff --git a/src/V3Options.cpp b/src/V3Options.cpp
index 7a85c84..5f2c5bd 100644
--- a/src/V3Options.cpp
+++ b/src/V3Options.cpp
@@ -864 +864 @@ string V3Options::getenvVERILATOR_ROOT() {
-    return V3Os::filenameCleanup(var);
+    return V3Os::filenameSlashPath(V3Os::filenameCleanup(var));
diff --git a/src/Verilator.cpp b/src/Verilator.cpp
index b1f0a9a..205af6a 100644
--- a/src/Verilator.cpp
+++ b/src/Verilator.cpp
@@ -840 +840 @@ static string buildMakeCmd(const string& makefile, const string& target) {
-    cmd << " -C " << v3Global.opt.makeDir();
+    cmd << " -C \"" << v3Global.opt.makeDir() << '"';
